.github/PULL_REQUEST_TEMPLATE/pull_request_template.md

```diff
@@ -0,0 +1,18 @@
+## Description
+<!--- Describe your changes -->
+
+## Motivation behind this PR?
+<!--- Why is the change required? Does it fix an existing issue, please link the issue. -->
+
+## What type of change is this?
+<!--- Bug Fix or Feature or Breaking Change i.e fix or feature that would cause existing functionality to not work as expected -->
+
+## Checklist
+<!-- If any particular item isn't necessary with your change, check it anyway so that the reviewer knows nothing is pending in the PR --> 
+
+- [ ] Added a changelog entry
+- [ ] Added unit tests
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] I have self-documented code my changes by adding docstring(s) and comment(s).
+- [ ] Current PR does not significantly decrease the code coverage and docstring coverage.
+- [ ] My code follows the style guidelines of this project.
```

Comment:
```suggestion
- [ ] My code follows the style guidelines of this project.

## Issue
<!--- All PRs must have a related issue. This way we can ensure that no one loses time working in something that does not needed to be done. -->

Closes <issue_number>
```

Reply:
@camilamaia added this part in the template but we might have to remove this when we finish the commit pattern design. As ideally, you should add a `Closes <issue link>` at the end of the commit message. 

That automatically references the related issue. But let's see how that goes when we finalize the commit pattern. Let's work with this for now. 😄

Reply:
@Pradhvan the same way works for the PRs! When closing the PR, it closes the issue too. 

I don't mind having this info in both places. It is easier for reviewers when the info is in the PR. But it is easy for checking history when the info is in the commit. What do you think?

Reply:
Sounds good to me. 👍🏾